WordPress/gutenberg
# [Mobile] -  RichText - Fix `undefined` onDelete callback

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/6925
- https://github.com/wordpress-mobile/WordPress-Android/pull/20973
- https://github.com/wordpress-mobile/WordPress-iOS/pull/23343

## What?
Fixes a crash when quickly deleting a Paragraph block and quickly focusing on a title and deleting it as well.

## Why?
To prevent a crash in this case scenario when editing content.

## How?
By adding a check for the `onDelete` callback in the RichText component.

## Testing Instructions

> [!NOTE]  
> Please use the following builds to test:
> - [iOS](https://github.com/wordpress-mobile/WordPress-iOS/pull/23343#issuecomment-2161185189)
> - [Android](https://github.com/wordpress-mobile/WordPress-Android/pull/20973#issuecomment-2161194183)

- Create a draft post with the following content:
  - Title: 1
  - Content: 1
- Save the draft
- Open the draft and quickly delete the content, then delete the title and focus on the first Paragraph block
- **Expect** the app to not crash 

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 
Before|After
-|-
<video src="https://github.com/WordPress/gutenberg/assets/4885740/358ea3c5-68cf-420d-b7fa-b67b5367958f" width=250 />|<video src="https://github.com/WordPress/gutenberg/assets/4885740/d6155bbc-37b8-4554-a94c-778099a8a5c9" width=250 />
